layers/cycle_paths.py

```diff
@@ -0,0 +1,149 @@
+    ensureEmptyTempDirectoryExists(tmp)
+
+    run(
+        [
+            "osmium",
+            "tags-filter",
+            osm_input,
+            "w/highway",
+            "-o",
+            f"{tmp}/cycle_paths.osm.pbf",
+        ]
+    )
+    gjPath = f"{tmp}/cycle_paths.geojson"
+    convertPbfToGeoJson(
+        f"{tmp}/cycle_paths.osm.pbf", gjPath, "linestring", includeOsmID=True
+    )
+
+    cleanUpGeojson(
+        gjPath, getProps, filterFeatures=lambda f: getProps(f["properties"]) != None
+    )
+
+    convertGeoJsonToPmtiles(f"{tmp}/cycle_paths.geojson", "output/cycle_paths.pmtiles")
+
+
+# If this has some kind of cycle-path, returns a dictionary with:
+#
+# - kind = track | lane | shared_use_segregated | shared_use_unsegregated
+# - osm_id
+# - direction = one-way | two-way | unknown
+#   (For cyclist traffic; unrelated to whether the path is a contraflow on a one-way street)
+# - width = number in meters | unknown
+#
+# If there's no cycle path, returns None
+def getProps(props):
+    if props.get("highway") == "cycleway":
+        kind = "track"
+        # Is it shared-use? Note "foot" may be missing (https://www.openstreetmap.org/way/849651126)
+        if props.get("foot") in ["yes", "designated"] or props.get("segregated"):
+            kind = getSharedUseKind(props)
+
+        return {
+            "kind": kind,
+            "osm_id": props["@id"],
+            "width": getSeparateWayWidth(props),
+            "direction": getSeparateWayDirection(props),
+        }
+    elif hasLane(props):
+        return {
+            "kind": "lane",
+            "osm_id": props["@id"],
+            "width": getLaneWidth(props),
+            "direction": getLaneDirection(props),
+        }
+    elif props["highway"] in ["footway", "pedestrian", "path", "track"]:
+        if props.get("bicycle") not in ["yes", "designated"]:
+            return None
+
+        return {
+            "kind": getSharedUseKind(props),
+            "osm_id": props["@id"],
+            "width": getSeparateWayWidth(props),
+            "direction": getSeparateWayDirection(props),
+        }
+    else:
+        return None
+
+
+def getSeparateWayWidth(props):
+    # TODO Handle suffixes / units
+    if props.get("width"):
+        return props["width"]
+    if props.get("est_width"):
+        return props["est_width"]
+    return "unknown"
+
+
+def getSeparateWayDirection(props):
+    oneway = props.get("oneway")
+    if oneway == "yes":
+        return "one-way"
+    elif oneway == "no":
+        return "two-way"
+    else:
+        return "unknown"
+
+
+def valueIndicatesLane(value):
+    return value and value not in ["no", "separate", "share_busway"]
+
+
+def hasLane(props):
+    # TODO Handle bicycle:lanes
+    for suffix in ["", ":left", ":right", ":both"]:
+        if valueIndicatesLane(props.get("cycleway" + suffix)):
+            return True
+    return False
+
+
+# If there are two lanes, the result could capture one or both of the lanes
```

Comment:
I suppose both here is not the width of both the lanes added together? I guess if left and right have different widths it's quite hard to get a good answer here